### DISCUSSION TITLE
Condition for Reactions

### DISCUSSION THREAD

**Discussion by slimshader** [codewriter-packages/UniMob#34] [Ideas]
Hey,

nothing serious as it is easy to add oneself but what do you think about adding additional overload to the Reaction, that would take a predicate upon a data and only then trigger a reaction? (equivalent f When() selector for Rx).

Often times I find myself opening reaction with "if" (especially for null checks), which makes it less declarative.

**Comment by vanifatovvlad** [comment]
I don’t really like RX operators, I think that a regular `if` is better because it more understandable

Can you give an example where condition is needed?
